### PR TITLE
Not caching Dataset classloader in DatasetAccessor when running in Explore Service

### PR DESCRIPTION
**Pull request by poornachandra**
This PR is for https://issues.cask.co/browse/CDAP-71

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java [hunk 1]
```diff
@@ -136,6 +136,7 @@ public void create(HttpRequest request, final HttpResponder responder,
     // Enable ad-hoc exploration of dataset
     // Note: today explore enable is not transactional with dataset create - CDAP-8
     try {
+      datasetExploreFacade.disableExplore(name);
```
Comment (jguery) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java]: So it just fails silently if there was no Hive table for ``name``?

File: cdap-explore/src/main/java/co/cask/cdap/hive/datasets/DatasetAccessor.java [hunk 2]
```diff
@@ -175,25 +193,35 @@ private static Dataset instantiate(@Nullable Configuration conf, String dsName)
...
-          throw new NullJobConfException();
-        }
-        classLoader = conf.getClassLoader();
-        dataset = firstLoad(framework, datasetName, classLoader);
-      } else {
-        dataset = framework.getDataset(datasetName, DatasetDefinition.NO_ARGUMENTS, classLoader);
+      
+      // Do not cache the Dataset classloader if running as part of Explore Service.
+      if (runContext == RunContext.EXPLORE_SERVICE) {
+        return framework.getDataset(datasetName, DatasetDefinition.NO_ARGUMENTS, null);
```
Comment (jguery) [inline cdap-explore/src/main/java/co/cask/cdap/hive/datasets/DatasetAccessor.java]: Shouldn't you at least ``conf.getClassloader()`` instead of ``null``?

**Comment by jguery** [comment]
Few questions, otherwise LGTM

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java (hunk 1 above)
Comment (poornachandra) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java]: Good point. It works, but it logs an error. I'll update as to not run the delete if table is not there.

File: cdap-explore/src/main/java/co/cask/cdap/hive/datasets/DatasetAccessor.java (hunk 2 above)
Comment (poornachandra) [inline cdap-explore/src/main/java/co/cask/cdap/hive/datasets/DatasetAccessor.java]: Good point, will change it.

**Comment by jguery** [comment]
LGTM

**Pull request merged**